code/parse/scripting.cpp

```diff
@@ -427,6 +428,11 @@ bool ConditionedHook::ConditionsValid(int action, object *objp, int more_data)
 									return false;
 								break;
 							}
+							case CHA_BEAMFIRE: {
+								if (!(stricmp(Weapon_info[more_data].name, scp->data.name) == 0))
```

Comment:
I think it would be better to move the `!` operator into the expression like this:
```C++
stricmp(Weapon_info[more_data].name, scp->data.name) != 0
```
Then it's clearer what the code should do (if that's even possible with the way `stricmp` works...).